### PR TITLE
Add Platform Tips Selection Box to the Add Funds Modal

### PR DESCRIPTION
**Pull request by SudharakaP**
This adds a Platform tips selection input to the Add Funds modal. 

Resolves https://github.com/opencollective/opencollective/issues/3652

Depends on: https://github.com/opencollective/opencollective-api/pull/5402

## Screen Cast

Platform Tips - Final

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
The code looks good to me.

It is a good practice to link the [API PR](https://github.com/opencollective/opencollective-api/pull/5402) from the Frontend PR in the description, especially since this PR is dependant on the other one.

**Comment by SudharakaP** [comment]
> The code looks good to me.
> 
> It is a good practice to link the [API PR](https://github.com/opencollective/opencollective-api/pull/5402) from the Frontend PR in the description, especially since this PR is dependant on the other one.

@Betree : Thanks for the review. Ah, sorry, I missed adding it. 😄

**Review by znarf — CHANGES_REQUESTED** [review]
In E2E test, can we add a check that proves that Platform Tips are properly recorded, ideally by checking on the Open Collective account page?

**Comment by SudharakaP** [comment]
> In E2E test, can we add a check that proves that Platform Tips are properly recorded, ideally by checking on the Open Collective account page?

@znarf : Yep, I've added the e2e tests as well for the whole `Add Funds` modal. 😄 

Note that I had to make the user a host admin on the API side (for E2E tests) since otherwise the `Add Funds` modal isn't viewable. Anyways, let me know if everything looks good. 😄

**Comment by SudharakaP** [comment]
There were some significant design changes to this with the latest design; therefore I am moving this back to draft. 😄

**Comment by SudharakaP** [comment]
Finished the first draft of the [new design](https://www.figma.com/file/ZQBMWhnGGtRWeIZknFW1eP/OC.com-Production-Ready-%E2%9C%85?node-id=10518%3A29405). Let me know what you think. Probably needs one more review. 😄 

Peek 2021-03-22 16-58

**Review by znarf — APPROVED** [review]
Looks good to me. Let's wait for a final review from @Betree.

File: test/cypress/integration/04-collective.test.js
Comment (Betree) [inline test/cypress/integration/04-collective.test.js:168]: I would rather add this in `test/cypress/integration/29-host.dashboard.test.js` and test the feature from the host dashboard. You'll find that tests are already setup there with a host admin account, which removes the need to add [this exception](https://github.com/opencollective/opencollective-api/pull/5402#discussion_r618355511) in the API.

**Review by Betree — COMMENTED** [review]
The frontend code looks good, my only blocker is what was reported in [the API](https://github.com/opencollective/opencollective-api/pull/5402#pullrequestreview-642129612).

File: test/cypress/integration/04-collective.test.js
Comment (SudharakaP) [reply test/cypress/integration/04-collective.test.js:168]: @Betree : The host dashboard doesn't have the actions dropdown if I am not mistaken. Therefore we cannot access the add fund modal there. Am I missing something? 🤔

Comment (Betree) [reply test/cypress/integration/04-collective.test.js:168]: There's no actions dropdown, but the add funds modal has its own button there: 
image

**Review by Betree — APPROVED** [review]
Would be great to have some error handling on addFunds / addPlatformTipToTransaction failures, but we can eventually add that later

File: test/cypress/integration/04-collective.test.js
Comment (SudharakaP) [reply test/cypress/integration/04-collective.test.js:36]: Ops, I forgot to remove this. Thanks for pointing it out. 👍🏽

File: test/cypress/integration/04-collective.test.js
Comment (SudharakaP) [reply test/cypress/integration/04-collective.test.js]: Ah, thanks for the explanation. That makes sense now. 😄

**Comment by SudharakaP** [comment]
> > Would be great to have some error handling on addFunds / addPlatformTipToTransaction failures, but we can eventually add that later
> 
> @SudharakaP We should iterate on that, it's important that we always display errors for mutations otherwise users have no way to know that it failed

@Betree : Ops, good that you remembered, I'll open a new issue for this. 👍🏽

**Pull request merged**
